kivymd/uix/button/button.py

```diff
@@ -903,333 +979,96 @@ def stop_elevation_anim(self):
-            ]
-
-
-class MDFlatButton(BaseFlatButton):
-    md_bg_color = ColorProperty([0.0, 0.0, 0.0, 0.0])
-
-
-class MDRectangleFlatButton(BaseRectangleFlatButton):
-    # TODO: If the user has set a custom text color, then when the application
-    #  color palette changes, the text color will also change to the current
-    #  color of the color scheme. Do need to preserve custom text colors when
-    #  changing the application palette?
-
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        Clock.schedule_once(self._remove_shadow)
+    _default_md_bg_color = None
+    _default_md_bg_color_disabled = None
+    _default_theme_text_color = "Custom"
+    _default_text_color = "PrimaryHue"
 
 
-class MDRectangleFlatIconButton(BaseRectangleFlatButton):
-    icon = StringProperty("android")
+class MDRectangleFlatButton(ButtonContentsText, BaseButton):
+    """A flat button with (by default) a primary color border and primary
+    color text.
     """
-    Button icon.
+    _default_line_color = None
+    _default_line_color_disabled = None
+    _default_theme_text_color = "Custom"
+    _default_text_color = "Primary"
 
-    :attr:`icon` is an :class:`~kivy.properties.StringProperty`
-    and defaults to `'android'`.
-    """
 
-    icon_color = ColorProperty(None)
+class MDRectangleFlatIconButton(ButtonContentsIconText, BaseButton):
+    """A flat button with (by default) a primary color border,
+    primary color text and a primary color icon on the left.
     """
-    Button icon color.
+    _default_line_color = None
+    _default_line_color_disabled = None
+    _default_theme_text_color = "Custom"
+    _default_text_color = "Primary"
 
-    :attr:`icon_color` is an :class:`~kivy.properties.ColorProperty`
-    and defaults to `None`.
-    """
-
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        Clock.schedule_once(self.remove_label)
-        Clock.schedule_once(self.set_icon_color)
-        Clock.schedule_once(self._remove_shadow)
-
-    def update_md_bg_color(
-        self, instance_theme_manager, name_palette: str
-    ) -> NoReturn:
-        self.text_color = self.theme_cls.primary_color
-        self.icon_color = self.theme_cls.primary_color
-
-    def set_icon_color(self, interval: Union[int, float]) -> NoReturn:
-        """Sets the icon color if no custom value is specified."""
-
-        if not self.icon_color:
-            self.icon_color = self.theme_cls.primary_color
-
-    def remove_label(self, interval: Union[int, float]) -> NoReturn:
-        self.remove_widget(self.ids.lbl_txt)
 
-
-class MDRoundFlatButton(MDFlatButton):
-    line_width = NumericProperty(1)
-    """
-    Line width for button border.
-
-    :attr:`line_width` is an :class:`~kivy.properties.NumericProperty`
-    and defaults to `1`.
+class MDRoundFlatButton(ButtonContentsText, BaseButton):
+    """A flat button with (by default) rounded corners, a primary
+    color border and primary color text.
     """
```

Comment:
"""
    A flat button with (by default) rounded corners, a primary
    color border and primary color text.
    """